python/src/spark_rapids_ml/tuning.py

```diff
@@ -121,27 +68,25 @@ def _fit(self, dataset: DataFrame) -> "CrossValidatorModel":
             subModels = [[None for j in range(numModels)] for i in range(nFolds)]
 
         datasets = self._kFold(dataset)
-        for i in range(nFolds):
-            # TODO, Need to get rid of cache when supporting transform/evaluate
-            # in a single pass for multiple models.
-            validation = datasets[i][1].cache()
-            train = datasets[i][0]
-
-            tasks = map(
-                inheritable_thread_target,
-                _parallelFitTasks(
-                    est, train, eva, validation, epm, collectSubModelsParam
-                ),
-            )
-            for j, metric, subModel in pool.imap_unordered(lambda f: f(), tasks):
-                metrics_all[i][j] = metric
-                if collectSubModelsParam:
-                    assert subModels is not None
-                    subModels[i][j] = subModel
-
-            validation.unpersist()
 
-        metrics, std_metrics = _gen_avg_and_std_metrics(metrics_all)
+        def singePassTask(
+            fold: int,
+        ) -> Tuple[int, List[float], Optional[List[_CumlModel]]]:
+            index_models = list(est.fitMultiple(datasets[fold][0], epm))
+            models = [model for _, model in index_models]
+            model = models[0]._combine(models)
+            metrics = model._transformEvaluate(datasets[fold][1], eva)
```

Comment:
Would it be possible for transformEvaluate to be a static method in the customized crossvalidator?

Reply:
Yeah, we can make the tansformEvaluate to be a static method in the crossvalidator, and it's just a proxy or switcher, finally it still needs to call a similar function from the specific model.

Reply:
Will leave it up to you to decide on this.